src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -141,7 +141,7 @@ private static Encoding GetFallbackEncoding()
             // For small streams, see if we can read the byte buffer directly.
             if (encoding.GetMaxCharCountOrThrowIfHuge(data) < LargeObjectHeapLimitInChars)
             {
-                if (TryGetBytesFromStream(data, out ArraySegment<byte> bytes) && bytes.Offset == 0)
+                if (TryGetBytesFromStream(data, out ArraySegment<byte> bytes) && bytes.Offset == 0 && bytes.Array is object)
```

Comment:
Is it possible to make an `ArraySegment` with a `null` `Array`? The xml docs on `ArraySegment` suggest that it's not possible.

Reply:
It's possible in the pathological case of `default(ArraySegment<T>)` but otherwise not possible. I'd be fine with using a `!` here instead of the extra check. I don't feel strongly either way.

Reply:
I'm guessing `Memory<byte>` would be the eventual goal.